lib/naughty/notification.lua

```diff
@@ -266,7 +266,7 @@ function notification:reset_timeout(new_timeout)
 
     self.timeout = new_timeout or self.timeout
 
-    if not self.timer.started then
+    if self.timer and not self.timer.started then
```

Comment:
what about adding first line to this method: `if not self.timer then return end`

and remove all further self.timer checks?

Reply:
If new_timeout is not nil or zero then `self.timeout = new_timeout` should be run.
And when it runs it would create the timer (or not, if new_timeout is zero).